lib/lightning/accounts.ex

```diff
@@ -24,33 +45,54 @@ defmodule Lightning.Accounts do
       # coveralls-ignore-stop
     end)
 
-    # Remove credentials
-    Credentials.list_credentials_for_user(id)
-    |> Enum.each(&Credentials.delete_credential/1)
-
-    # Revoke access to projects
+    # Delete all project_users
     Projects.get_projects_for_user(%User{id: id})
     |> Repo.preload(:project_users)
-    |> Enum.each(fn p ->
-      Projects.update_project(
-        p,
-        %{
-          "project_users" => %{
-            "0" => %{
-              "delete" => "true",
-              "user_id" => id,
-              "id" =>
-                Enum.find(p.project_users, fn pu -> pu.user_id == id end)
-                |> Map.get(:id)
-            }
-          }
-        }
-      )
+    |> Enum.map(fn p ->
+      {p, %{"project_users" => project_users_to_params(p.project_users, id)}}
     end)
+    |> Enum.each(fn {project, param} ->
+      Projects.update_project(project, param)
+    end)
+
+    # Prep credentials for deletion
+    credentials_for_user = Credentials.list_credentials_for_user(id)
 
-    User
-    |> Repo.get(id)
-    |> delete_user()
+    user_credentials_ids =
+      Enum.map(credentials_for_user, fn credential -> credential.id end)
+
+    project_credentials_for_user =
+      from(project_credential in Lightning.Projects.ProjectCredential,
+        where: project_credential.credential_id in ^user_credentials_ids
+      )
+      |> Repo.all()
+
+    user_project_credentials_ids =
+      Enum.map(project_credentials_for_user, fn project_credential ->
+        project_credential.id
+      end)
+
+    # Set null all jobs using credentials
+    :ok =
+      from(job in Lightning.Jobs.Job,
+        where: job.project_credential_id in ^user_project_credentials_ids
+      )
+      |> Repo.all()
+      |> Enum.each(fn job ->
+        Lightning.Jobs.update_job(job, %{credential_id: nil})
```

Comment:
Jobs don't have a `credential_id`. They have a credential via a `project_credential_id`. 

I'm a bit concerned the tests didn't pick this up. 

There could also be a cascade rule that makes this update unnecessary, i.e. the deletion of a project credential sets `project_credential_id` to `nil`.

Reply:
I confirm there's a nillify rule. Thanks